### ISSUE TITLE
Is it possible to request diagnostics?

### ISSUE THREAD

**Issue by eriktobben** [filipp/gsxlib#51]
Hi!

Is it possible to request a diagnostic from diagnostics.apple.com (AST2)?

In GSX there is now a button "Request Diagnostics" and you are redirected to diagnostics.apple.com with the serial number prefilled.

**Comment by Kuchiru** [comment]
Depends what you're asking.
If you want to know if it's possible with gsxlib at this time, the answer is no.
If you want to know if it's possible with the API, yes it's possible.

**Comment by eriktobben** [comment]
I am asking if its possible with gsxlib. Is there plans to implement this in the near future? :)

**Comment by bastida117** [comment]
i have write some lines of code to get all the event numbers and then to get the details of that event numbers, but so far, im stuck cause:

1) if i don't provide an event number i get only the last diagnostic executed in the device

2) if i provide a event number i get a gsx.diags.022 error from gsx but is not documented under the GSX WEB SERVICES that error and the error in response basically said that i must provide a serial number (which im providing), a ship-to (also providing) and a repair number or diagnostic event number (which also im providing)

So in conclusion, adding the third parameter breaks the thing, and omitting the last parameter only brings the last diagnostic executed which is not very useful cause at the end if you run MRI, VST ,and many other you will only get the info about the last one

**Comment by Kuchiru** [comment]
@bastida117 I tested the behaviour you are describing and i can confirm what you're experiencing.
However, the error states that you should just use one of the fields, not as much as possible.

So only providing the diagnosticEventNumber or the serialNumber, not both, would give you the results you're looking for.
